src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -36,6 +36,9 @@ internal static class ThreadPoolGlobals
         public static volatile bool vmTpInitialized;
         public static bool enableWorkerTracking;
 
+        public const int wsqNowActive = 2; // Require at least 2 full observations of empty wsq to transition to inactive
```

Comment:
EnsureThreadRequested is done after an item is added, I believe that would prevent the race by causing a thread to wake up if necessary to scan the queues